### PR TITLE
Add support for uPickle

### PR DESCRIPTION
**Pull request by justinhj**
Adds uPickle as an alternative to Circe for json coding/decoding

### CODE REVIEW

File: json/upickle/src/main/scala/tapir/upickle/TapirJsonuPickle.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import scala.util.{Try, Success, Failure}
+import tapir.Codec.JsonCodec
+import tapir.DecodeResult.{Error, Value}
+import tapir._
+import upickle.default.{ReadWriter => RW}
+import upickle.default.{read, write}
+
+trait TapirJsonuPickle {
+
+  implicit def encoderDecoderCodec[T: RW: SchemaFor, M]: JsonCodec[T] = new JsonCodec[T] {
```
Comment (guersam) [inline json/upickle/src/main/scala/tapir/upickle/TapirJsonuPickle.scala]: What is the type parameter `M` for?

File: version.sbt [hunk 2]
```diff
@@ -1 +1 @@
-version in ThisBuild := "0.9.2-SNAPSHOT"
+version in ThisBuild := "0.9.2-JUSTIN-SNAPSHOT"
```
Comment (guersam) [inline version.sbt]: It probably should be rolled back.

File: json/upickle/src/main/scala/tapir/upickle/TapirJsonuPickle.scala [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package tapir.json.upickle
```
Comment (guersam) [inline json/upickle/src/main/scala/tapir/upickle/TapirJsonuPickle.scala]: The file path does not match the package name. Should be `tapir/json/upickle` I guess

File: json/upickle/src/main/scala/tapir/upickle/TapirJsonuPickle.scala (hunk 1 above)
Comment (justinhj) [reply json/upickle/src/main/scala/tapir/upickle/TapirJsonuPickle.scala]: Removed. That must have been from an earlier iteration of the code :)

File: json/upickle/src/main/scala/tapir/json/upickle/TapirJsonuPickle.scala [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package tapir.json.upickle
+
+import java.nio.charset.StandardCharsets
+import scala.util.{Try, Success, Failure}
+import tapir.Codec.JsonCodec
+import tapir.DecodeResult.{Error, Value}
+import tapir._
+import upickle.default.{ReadWriter => RW}
```
Comment (guersam) [inline json/upickle/src/main/scala/tapir/json/upickle/TapirJsonuPickle.scala]: Is this alias necessary? `ReadWriter` is not too long to read and doesn't seem like shadowing anything.

Comment (justinhj) [reply json/upickle/src/main/scala/tapir/json/upickle/TapirJsonuPickle.scala]: You're quite right. Fixed

**Comment by ghostbuster91** [comment]
This looks great :) Could you add a test for it inside upickle module?

**Comment by justinhj** [comment]
> This looks great :) Would you consider adding some information to docs?

Sure I'd be happy to add docs and tests

**Comment by justinhj** [comment]
@ghostbuster91 I have updated the documentation with respect to uPickle support and added tests in the upickle module.

**Comment by adamw** [comment]
Released in 0.9.2, thanks!

**Comment by justinhj** [comment]
Awesome, thanks @adamw

**Pull request merged**